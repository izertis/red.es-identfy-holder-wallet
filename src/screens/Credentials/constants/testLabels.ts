export const TEST_LABELS = {
  ActionButton: "ActionButton",
  DropDownSelect: "DropDownSelect",
  DropDownView: "DropDownView",
}
