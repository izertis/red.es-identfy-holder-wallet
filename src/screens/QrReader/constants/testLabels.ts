export const TEST_LABELS = {
    BACKGROUND_IMAGE: "background-image"
  }