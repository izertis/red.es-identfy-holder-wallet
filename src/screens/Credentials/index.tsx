import { useCallback, useRef, useState, useEffect } from 'react'
import { StatusBar } from 'react-native'
import CredentialStyled from './styles'
import { NavigationInjectedProps } from 'react-navigation'
import Placeholder from '../../components/Placeholder'
import CredentialsI18nKeys from './i18n/keys'
import { SCREEN } from '../../constants/screens'
import { useTranslation } from 'react-i18next'
import { getCredentialsList } from '../../utils/keychain'
import CredentialList from './components/CredentialList'
import ButtonScanner from './components/ButtonScanner'
import StateWrapper from '../../components/wrappers/StateWrapper'
import { StateWrapperRef } from '../../components/wrappers/StateWrapper/type'
import { useFocusEffect } from '@react-navigation/native'

interface Props extends NavigationInjectedProps { }

const Credentials = (props: Props) => {
  const { t } = useTranslation(SCREEN.Credentials)
  const childRef = useRef<StateWrapperRef>()

  const reloadCredentials = () => {
    childRef.current?.reload?.()
  }

  useFocusEffect(
    useCallback(() => {
      reloadCredentials()
    }, []),
  )

  const onHandleCredentials = async () => {
    const credentials = await getCredentialsList()
    return credentials
  }

  const placeholderParams = {
    title: t(CredentialsI18nKeys.PlACEHOLDER_TITLE),
    subtitle: t(CredentialsI18nKeys.PlACEHOLDER_TEXT),
  }

  return (
    <CredentialStyled.ContentView>
      <StatusBar backgroundColor='transparent' barStyle='light-content' />
      <StateWrapper
        ref={childRef}
        initialFunction={onHandleCredentials}
        emptyStateCondition={(credentials) => !credentials.length}
        renderEmptyState={
          <Placeholder
            placeholderTitle={placeholderParams?.title}
            placeholderText={placeholderParams?.subtitle}
          />
        }
        renderData={(credentials) => (
          <CredentialList
            credentials={credentials}
          />
        )}
      />
      <ButtonScanner
        onPress={() => {
          props.navigation.navigate('QrReader')
        }}
      />
    </CredentialStyled.ContentView>
  )
}

export default Credentials