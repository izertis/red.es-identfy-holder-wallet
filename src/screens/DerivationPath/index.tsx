import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Card } from 'react-native-paper'
import { RootStackScreenProps } from '../../../types'
import Button from '../../components/atomic_components/Button'
import PressToCopy from '../../components/PressToCopy'
import { SCREEN } from '../../constants/screens'
import DerivationPathI18nKeys from './i18n/keys'
import DerivationPathStyled from './styles'

const DerivationPath = ({ route, navigation }: RootStackScreenProps<'DerivationPath'>) => {
	const { t } = useTranslation(SCREEN.DerivationPath)

	const { derivationPath }: any = route.params

	const [isSeedBlur, setIsSeedBlur] = useState<boolean>(true)

	return (
		<DerivationPathStyled.MainContainer>
			<DerivationPathStyled.BackButton onPress={() => navigation.pop()}>
				{t(DerivationPathI18nKeys.GO_BACK)}
			</DerivationPathStyled.BackButton>
			<DerivationPathStyled.Container>
				<DerivationPathStyled.Title testID='title'>
					{t(DerivationPathI18nKeys.TITLE)}
				</DerivationPathStyled.Title>
				<DerivationPathStyled.Subtitle>{t(DerivationPathI18nKeys.SUBTITLE)}</DerivationPathStyled.Subtitle>

				{isSeedBlur ? (
					<DerivationPathStyled.PhraseContainer onPress={() => setIsSeedBlur(!isSeedBlur)}>
						<DerivationPathStyled.MainCard justifyContent='flex-end' alignItems='center'>
							<Card.Content>
								<DerivationPathStyled.HiddenTitle>
									{t(DerivationPathI18nKeys.HIDDEN_TITLE)}
								</DerivationPathStyled.HiddenTitle>
							</Card.Content>
						</DerivationPathStyled.MainCard>
					</DerivationPathStyled.PhraseContainer>
				) : (
					<DerivationPathStyled.PhraseContainer>
						<PressToCopy onPress={() => setIsSeedBlur(!isSeedBlur)} value={derivationPath}>
							<DerivationPathStyled.MainCard>
								<Card.Content>
									<DerivationPathStyled.Content>{derivationPath}</DerivationPathStyled.Content>
								</Card.Content>
							</DerivationPathStyled.MainCard>
						</PressToCopy>
					</DerivationPathStyled.PhraseContainer>
				)}
				{!isSeedBlur && (
					<DerivationPathStyled.CopyText>
						{t(DerivationPathI18nKeys.LONG_PRESS)}
					</DerivationPathStyled.CopyText>
				)}

				<DerivationPathStyled.ContainerBottom>
					<Button
						onPress={() => {
							navigation.navigate('NetworkAuth')
						}}
					>
						{t(DerivationPathI18nKeys.BUTTON)}
					</Button>
				</DerivationPathStyled.ContainerBottom>
			</DerivationPathStyled.Container>
		</DerivationPathStyled.MainContainer>
	)
}

export default DerivationPath
