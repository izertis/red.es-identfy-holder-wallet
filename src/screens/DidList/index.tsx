import React, { useEffect, useState } from "react"
import { useTranslation } from "react-i18next"
import { RootStackScreenProps } from "../../../types"
import { SCREEN } from "../../constants/screens"
import { getDidList } from "../../utils/keychain"
import DidCard from "./components/DidCard"
import PressToCopy from "../../components/PressToCopy"
import DerivationPathStyled from "../DerivationPath/styles"
import DidListI18nKeys from "./i18n/keys"
import DidListStyled from "./styles"

const DidList = ({ navigation }: RootStackScreenProps<"DidList">) => {
  const { t } = useTranslation(SCREEN.DidList)
  const [didList, setDidList] = useState<{ network: string, entity: string, did: string }[]>([])

  useEffect(() => {
    ; (async () => {
      const didList: any = await getDidList()
      setDidList(didList)
    })()
  }, [])

  const hasDIDs = didList.some(({ did }) => !!did)

  return (
    <DidListStyled.MainContainer>
      {hasDIDs ? (
        <>
          {didList.filter(did => !!did.did).map(({ did, network, entity }, index) => (
            <PressToCopy value={did} key={index}>
              <DidCard key={`didCard-${index}`} did={did} network={network} entity={entity} />
            </PressToCopy>
          ))}
          <DidListStyled.TextContainer>
            <DerivationPathStyled.CopyText>
              {t(DidListI18nKeys.PRESS_TO_COPY)}
            </DerivationPathStyled.CopyText>
          </DidListStyled.TextContainer>
        </>
      ) : (
        <DidListStyled.TextContainer>
          <DerivationPathStyled.CopyText>
            {t(DidListI18nKeys.NO_DID_CREATED)}
          </DerivationPathStyled.CopyText>
        </DidListStyled.TextContainer>
      )}
      {didList.filter((elm) => elm.did !== null).length !== 3 ? (
        <DidListStyled.ContainerBottom>
          <DidListStyled.Button
            onPress={() => {
              navigation.navigate('NetworkAuth')
            }}
          >
            {t(DidListI18nKeys.ADD_NEW_DID)}
          </DidListStyled.Button>

        </DidListStyled.ContainerBottom>
      ) : null}
    </DidListStyled.MainContainer>
  )
}

export default DidList
