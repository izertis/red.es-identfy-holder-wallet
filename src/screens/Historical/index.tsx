import { useCallback, useRef } from 'react'
import { NavigationInjectedProps } from 'react-navigation'
import Placeholder from '../../components/Placeholder'
import { SCREEN } from '../../constants/screens'
import { useTranslation } from 'react-i18next'
import { checkAndLogExpiredCredentials, getActionLogs } from '../../utils/keychain'
import StateWrapper from '../../components/wrappers/StateWrapper'
import { StateWrapperRef } from '../../components/wrappers/StateWrapper/type'
import { useFocusEffect } from '@react-navigation/native'
import HistoricalStyled from './styles'
import HistoricalList from './components/HistoricalList'
import HistoricalI18nKeys from './i18n/keys'
import { StatusBar } from 'react-native'

interface Props extends NavigationInjectedProps { }

const Historical = (props: Props) => {
  const { t } = useTranslation(SCREEN.Historical)
  const childRef = useRef<StateWrapperRef>()

  useFocusEffect(
    useCallback(() => {
      checkAndLogExpiredCredentials()
    }, []),
  )

  const onHandleCredentials = async () => {
    const logs = await getActionLogs()
    return logs
  }

  const placeholderParams = {
    title: t(HistoricalI18nKeys.PLACEHOLDER_TITLE),
    subtitle: t(HistoricalI18nKeys.PLACEHOLDER_TEXT),
  }

  return (
    <HistoricalStyled.ContentView>
      <StateWrapper
        ref={childRef}
        initialFunction={onHandleCredentials}
        emptyStateCondition={(logs) => !logs.length}
        renderEmptyState={
          <Placeholder
            placeholderTitle={placeholderParams?.title}
            placeholderText={placeholderParams?.subtitle}
          />
        }
        renderData={(logs) => (
          <HistoricalList logs={logs} />
        )}
      />
    </HistoricalStyled.ContentView>
  )
}

export default Historical
