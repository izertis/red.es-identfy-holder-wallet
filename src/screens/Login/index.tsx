import { useContext, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { RootStackScreenProps } from '../../../types'
import { SCREEN } from '../../constants/screens'
import { MessageContext } from '../../context/UserMessage.context'
import LocalStorageService, {
  STORAGE_KEYS,
} from '../../services/LocalStorage.service'
import { checkPin } from '../../utils/keychain'
import RegisterI18nKeys from './i18n/keys'
import LoginStyled from './styles'
import { View } from '../../components/Themed'
import { executeCallbackFunction } from '../../utils/storeCallback'
import { StatusBar } from 'expo-status-bar'
import { ColorKeys, getThemeColor } from '../../constants/Colors'
import { Keyboard, Platform } from 'react-native'

const Login = ({ route, navigation }: RootStackScreenProps<'Login'>) => {
  const { t } = useTranslation(SCREEN.Login)
  const { callback }: any = route?.params ?? {}
  const { showMessage } = useContext(MessageContext)
  const [isLoading, setIsLoading] = useState<boolean>(false)
  const [pin, setPin] = useState<string>('')

  const login = async () => {
    Keyboard.dismiss()
    const IS_DID_CREATED = await LocalStorageService.getBool(
      STORAGE_KEYS.IS_DID_CREATED
    )
    await LocalStorageService.storeData(STORAGE_KEYS.PIN, pin)
    setPin('')
    if (IS_DID_CREATED) {
      try {
        callback
          ? await executeCallbackFunction(callback.id, callback.params)
          : navigation.replace('Root')
      } catch (error) { }
    } else {
      navigation.replace('NetworkAuth')
    }
    setPin('')
    setIsLoading(false)
  }

  const showError = (): void => {
    const content = t(RegisterI18nKeys.ERROR)
    showMessage({ content, type: 'error' })
    setPin('')
    setIsLoading(false)
  }

  return (
    <LoginStyled.MainContainer>
      <StatusBar backgroundColor='transparent' style='auto' />
      <View>
        <LoginStyled.RegisterButton
          onPress={() => navigation.navigate('Register')}
        >
          {t(RegisterI18nKeys.REGISTER_LINK)}
        </LoginStyled.RegisterButton>

        <LoginStyled.Title testID="title">
          {t(RegisterI18nKeys.TITLE)}
        </LoginStyled.Title>
        <LoginStyled.InputsContainer>
          <LoginStyled.InputStyled
            disabled={isLoading}
            value={pin}
            keyboardType="decimal-pad"
            secureTextEntry
            onChangeText={(value: string) =>
              setPin(value.replace(/[^0-9]/g, ''))
            }
            placeholder={t(RegisterI18nKeys.PIN_LABEL) || ''}
            onSubmitEditing={async () => ((await checkPin(pin)) ? login() : showError())}
            returnKeyType={Platform.OS === 'android' ? 'send' : 'done'}
          />
        </LoginStyled.InputsContainer>
      </View>
      <LoginStyled.ContainerBottom>
        {isLoading ? <LoginStyled.SplashActivityIndicator color={getThemeColor(ColorKeys.text)} size={35} /> : null}
        <LoginStyled.Button
          testID="startButton"
          disabled={pin === '' || isLoading}
          style={{ flexDirection: 'row' }}
          onPress={async () => {
            setIsLoading(true);
            ((await checkPin(pin)) ? login() : showError())
          }}
        >
          {t(RegisterI18nKeys.BUTTON)}
        </LoginStyled.Button>
      </LoginStyled.ContainerBottom>
    </LoginStyled.MainContainer>
  )
}

export default Login
