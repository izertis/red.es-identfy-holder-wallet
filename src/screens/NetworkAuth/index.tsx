import { useContext, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { KeyboardAvoidingView, View } from 'react-native'
import { Text } from 'react-native-paper'
import { RootStackScreenProps } from '../../../types'
import { SCREEN } from '../../constants/screens'
import { MessageContext } from '../../context/UserMessage.context'
import NetworkAuthI18nKeys from './i18n/keys'
import NetworkAuthStyled from './styles'
import { createDidLacchain } from '../../services/lacchain/createDid'
import { createDidEbsi } from '../../services/ebsi/createDid'
import { getKeychainDataObject } from '../../utils/keychain'
import { createDidAlastria } from '../../services/alastria/createDid'
import { StatusBar } from 'expo-status-bar'
import { ColorKeys, getThemeColor } from '../../constants/Colors'

interface Checkbox {
	entity: string,
	label: string
	subLabel: string
	checked: boolean
	action: () => Promise<void>
}

const NetworkAuth = ({ navigation }: RootStackScreenProps<'NetworkAuth'>): JSX.Element => {
	const { t } = useTranslation(SCREEN.NetworkAuth)

	const { showMessage } = useContext(MessageContext)
	const [dids, setDids] = useState<any>({})
	const [isLoading, setIsLoading] = useState<boolean>(false)

	useEffect(() => {
		const fetchData = async () => {
			const keychain = await getKeychainDataObject()
			setDids(keychain?.did || {})
		}
		fetchData()
	}, [])

	const checkboxes: Checkbox[] = [
		{
			entity: 'alastria',
			label: t(NetworkAuthI18nKeys.CHECKBOX_ALASTRIA),
			subLabel: t(NetworkAuthI18nKeys.CHECKBOX_ALASTRIA_SUB),
			checked: dids.alastria?.AlastriaT || false,
			action: async () => {
				return await createDidAlastria()
			},
		},
		{
			entity: 'lacchain',
			label: t(NetworkAuthI18nKeys.CHECKBOX_LACCHAIN),
			subLabel: t(NetworkAuthI18nKeys.CHECKBOX_LACCHAIN_SUB),
			checked: dids.lacchain || false,
			action: async () => {
				return await createDidLacchain()
			},
		},
		{
			entity: 'ebsi',
			label: t(NetworkAuthI18nKeys.CHECKBOX_EBSI),
			subLabel: t(NetworkAuthI18nKeys.CHECKBOX_EBSI_SUB),
			checked: dids.ebsi || false,
			action: async () => {
				return await createDidEbsi()
			},
		},
	]

	const [checkboxStates, setCheckboxStates] = useState<{ [key: string]: boolean }>(
		checkboxes.reduce((acc: any, curr) => {
			acc[curr.label] = curr.checked
			return acc
		}, {})
	)

	const handleCreateDID = async (): Promise<void> => {
		setIsLoading(true)
		let isSelected = false
		try {
			if (checkboxStates[t(NetworkAuthI18nKeys.CHECKBOX_ALASTRIA)]) {
				await checkboxes[0].action()
			}
			if (checkboxStates[t(NetworkAuthI18nKeys.CHECKBOX_LACCHAIN)]) {
				await checkboxes[1].action()
			}
			if (checkboxStates[t(NetworkAuthI18nKeys.CHECKBOX_EBSI)]) {
				await checkboxes[2].action()
			}

			for (const key in checkboxStates) {
				if (checkboxStates[key] === true) {
					isSelected = true
					break
				}
			}
			isSelected && showMessage({ content: t?.(NetworkAuthI18nKeys.SUCCESS), type: "success" })

		} catch (error) {
			showMessage({ content: t?.(NetworkAuthI18nKeys.ERROR), type: "error" })
		} finally {
			setIsLoading(false)
			navigation.reset({
				index: 0,
				routes: [{ name: 'Root' }],
			})
		}
	}

	const handleCheckboxChange = (label: string): void => {
		setCheckboxStates({
			...checkboxStates,
			[label]: !checkboxStates[label],
		})
	}
	const isCreateDidNeeded = Object.values(checkboxStates).some((checked) => !!checked)

	return (
		<KeyboardAvoidingView>
			<StatusBar backgroundColor='transparent' style='auto' />
			<NetworkAuthStyled.MainContainer>
				<NetworkAuthStyled.GoBackButtonContainer>
					{navigation.canGoBack() &&
						<NetworkAuthStyled.GoBackButton onPress={() => navigation.pop()}>
							{t(NetworkAuthI18nKeys.GO_BACK)}
						</NetworkAuthStyled.GoBackButton>
					}

				</NetworkAuthStyled.GoBackButtonContainer>

				<NetworkAuthStyled.Title testID='title'>{t(NetworkAuthI18nKeys.TITLE)}</NetworkAuthStyled.Title>

				<NetworkAuthStyled.Subtitle>{t(NetworkAuthI18nKeys.SUBTITLE)}</NetworkAuthStyled.Subtitle>

				{checkboxes.map((checkbox: Checkbox, index: number) => (
					<NetworkAuthStyled.CheckBoxContainer key={index}>
						<NetworkAuthStyled.CheckBox
							checked={checkboxStates[checkbox.label] || dids[checkbox.entity]}
							disabled={!!dids[checkbox.entity] || isLoading}
							onPress={() => handleCheckboxChange(checkbox.label)}
						/>
						<View>
							<Text style={{ fontWeight: 'bold' }}>{checkbox.label}</Text>
							<Text>{checkbox.subLabel}</Text>
						</View>
					</NetworkAuthStyled.CheckBoxContainer>
				))}

				<NetworkAuthStyled.Subtitle>{t(NetworkAuthI18nKeys.FOOTER_TEXT)}</NetworkAuthStyled.Subtitle>

				<NetworkAuthStyled.ContainerBottom>
					{isLoading ? <NetworkAuthStyled.SplashActivityIndicator color={getThemeColor(ColorKeys.text)} size={35} /> : null}
					<NetworkAuthStyled.Button
						testID='create-did-button'
						disabled={isLoading}
						onPress={() => handleCreateDID()}
					>
						{isLoading
							? t(NetworkAuthI18nKeys.LOADING)
							: isCreateDidNeeded
								? t(NetworkAuthI18nKeys.BUTTON)
								: t(NetworkAuthI18nKeys.CONTINUE)
						}
					</NetworkAuthStyled.Button>
				</NetworkAuthStyled.ContainerBottom>
			</NetworkAuthStyled.MainContainer>
		</KeyboardAvoidingView>
	)
}

export default NetworkAuth
