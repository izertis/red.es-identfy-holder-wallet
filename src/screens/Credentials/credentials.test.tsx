import React from "react"
import { render } from "@testing-library/react-native"

import Credential from "."
import { TEST_LABELS } from "./constants/testLabels"
import i18next from "../../../i18n.config";
import { SCREEN } from "../../constants/screens";
import localeES from "./i18n/es";


const mockedNavigation: any = {
  navigate: jest.fn(),
  route: jest.fn(),
}
describe("<Credentials />", () => {
  const ScreenName = SCREEN.Credentials;
  
  // Adds a resource bundle for the Spanish language to i18next with the translations for the Register screen
  i18next.addResourceBundle("es", ScreenName, localeES);
  it("should render ", () => {
    const result = render(<Credential {...mockedNavigation} />)

    expect(result.toJSON()).toMatchSnapshot()
  })
  it("Should renders correctly", () => {
    const { getAllByRole, getByTestId } = render(
      <Credential {...mockedNavigation} i18next={i18next}/>
    )

    const ActionButton = getByTestId(TEST_LABELS.ActionButton)

    expect(ActionButton).toBeDefined()
  })
})
