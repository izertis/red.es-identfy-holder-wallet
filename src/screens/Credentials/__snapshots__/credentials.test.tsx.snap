// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Credentials /> should render  1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Array [
        Object {
          "height": "92%",
        },
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityRole="progressbar"
        accessibilityState={
          Object {
            "busy": true,
          }
        }
        accessible={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 25,
              "opacity": 1,
              "width": 25,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 25,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 25,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 12.5,
                    "overflow": "hidden",
                    "width": 25,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 25,
                      "transform": Array [
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "rotate": "-165deg",
                        },
                      ],
                      "width": 25,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 12.5,
                        "overflow": "hidden",
                        "width": 25,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "borderColor": "#eb5b5b",
                          "borderRadius": 12.5,
                          "borderWidth": 2.5,
                          "height": 25,
                          "width": 25,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "height": 25,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 25,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 12.5,
                    "overflow": "hidden",
                    "top": 12.5,
                    "width": 25,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 25,
                      "transform": Array [
                        Object {
                          "translateY": -12.5,
                        },
                        Object {
                          "rotate": "345deg",
                        },
                      ],
                      "width": 25,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 12.5,
                        "overflow": "hidden",
                        "width": 25,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "borderColor": "#eb5b5b",
                          "borderRadius": 12.5,
                          "borderWidth": 2.5,
                          "height": 25,
                          "width": 25,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 16,
          "bottom": -20,
          "elevation": 6,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
          "position": "absolute",
          "right": 15,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#eb5b5b",
          "borderRadius": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="ActionButton-container-outer-layer"
    >
      <View
        collapsable={false}
        size={82}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#eb5b5b",
            "borderColor": "transparent",
            "borderRadius": 16,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "justifyContent": "center",
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="ActionButton-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 16,
              },
            ]
          }
          testID="ActionButton"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#eb5b5b",
                  "paddingVertical": 16,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginLeft": 12,
                    "marginRight": -4,
                  },
                  Object {
                    "marginLeft": 16,
                    "marginRight": -16,
                  },
                  false,
                ]
              }
              testID="ActionButton-icon-container"
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#000",
                        "fontFamily": "Muli",
                        "letterSpacing": 0.57,
                      },
                    ],
                  ],
                ]
              }
              testID="ActionButton-text"
            >
              BUTTON_SCAN
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
